src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,14 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportGenerator;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder("fruit,quantity\n");
+        Storage.getFruits().forEach((fruit, quantity) -> report.append(fruit)
+                .append(",").append(quantity).append("\n"));
```

Comment:
Direct access to the `Storage.getFruits()` method violates the encapsulation principle. Instead of accessing the storage directly, consider creating a service that provides the necessary data. This way, you can avoid coupling your report generation logic tightly to the storage implementation.